src/utils/config.c

```diff
@@ -357,3 +361,112 @@ bool conf_is_list(config_setting_t *conf, const char *file, size_t line,
     }
     return true;
 }
+
+/*
+ * config_group_apply
+ *      Apply a function to a group. Optional `arg` can be used to pass user
+ *      data into the specified function.
+ */
+void config_group_apply(const config_setting_t *options, group_func func, void *arg)
+{
+    int noptions;
+
+    if (!options)
+        return;
+
+    noptions = config_setting_length(options);
+
+    for (int i = 0; i < noptions; ++i)
+    {
+        config_setting_t *o;
+        const char *key;
+        const char *value;
+
+        o = config_setting_get_elem(options, i);
+        key = config_setting_name(o);
+        value = config_setting_get_string(o);
+
+        func(key, value, arg);
+    }
+}
+
+/*
+ * config_create_path
+ *      Creates path nodes except the leaf one. The path must be specified
+ *      in the form "one/two/three", e.i. path parts are separated with the '/'
+ *      symbol
+ */
+config_setting_t *config_create_path(config_setting_t *parent,
+                                     const char *path,
+                                     int type)
+{
+    char   *sep;
+    config_setting_t *cur = parent;
+    char    buf[512];
+    char   *prefix = buf;
+
+    /* TODO: put an assert on strlen <= 512 */
+    strcpy(buf, path);
+
+    while ((sep = strchr(prefix, PATH_SEPARATOR)) != NULL)
+    {
+        config_setting_t *found;
+        *sep = '\0';
```

Comment:
`path` is a constant string, we drop the `const` qualifier above assiging it to `prefix` and then edit it here.
This is an invitation to undefined behaviour.

Reply:
but we copy it to the buffer and assign the copied version of string to `prefix`. Or do i miss something?

Reply:
Ildar is right on this one, prefix is a writable copy of a const char, not a const char.
Technically, buf[512] should decay into a char * on function call and we'd not need to copy its address into prefix (that's how line 410 works for that matter).
Both nitpicks that don't need addressing.